### ISSUE TITLE
Dataset download issue for PDBbind Dataset

### ISSUE THREAD

**Issue by saurabhclusterone** [deepchem/deepchem#1430]
Hello All,

I am working on the pdbbind_tf.py example. I want to make this example run in distributed mode.

But, everytime I run the code, the code fails with following error:

TypeError: 'NoneType' object is not iterable

The reason being I don't see the dataset file getting downloaded from the Amazon S3 bucket URL embedded in the code. I tried to copy paste the URL, it still would not work.

Can you please tell me how to get the dataset for this code and model to get working?

**Comment by VIGS25** [comment]
It must be because of the indentation error in load_pdbbind_grid().
You could either copy line 134 and add it after line 76 or you could just move lie 134 up by one level of indentation.

**Comment by VIGS25** [comment]
This must be because of the indentation error in line 134 of `load_pdbbind_grid()`.
I would recommend copying the line 134 and adding it after line 76, to get it running for now.

Will raise a PR for this ASAP as well.

**Comment by li-aolong** [comment]
@VIGS25 In line 70 of `load_pdbbind_grid()`, why `transformers = []`. Is this a bug?

**Comment by saurabhclusterone** [comment]
@VIGS25 

The indentation was fixed by me already before running on locally. The error is also different and does not seem related to code indentation.

Can you please look into this again, can you run the code on your local please?

**Comment by VIGS25** [comment]
@saurabhclusterone: When I first ran the code (my indentation was fixed), it ran normally without any errors. I removed the indentation fix and upon running it again, had the same error message you had. 

I'm running it again now, will let you know.

**Comment by saurabhclusterone** [comment]
@VIGS25 

Were you able to run successfully?

**Comment by VIGS25** [comment]
Yes, have a look a the screenshot above.
The first one, is the one which runs normally and the second one is with the error you mention

**Comment by saurabhclusterone** [comment]
@VIGS25 

Did you get to merge your changes with the actual example code in repo.

Because I ran the code again from the repo, and it still gives same error.

**Comment by VIGS25** [comment]
Run `python setup.py install` and re-run your script and let me know?

**Comment by saurabhclusterone** [comment]
Hello @VIGS25 ,

I tried different ways, including your way of running setup.py in the local repo.

But the error still persist.

One important info: I am running in python 2.7 conda environment. I have run many other deepchem code in same environment before without any problem.

**Comment by VIGS25** [comment]
@saurabhclusterone: I'm not sure what is happening then, as I am able to download the datasets even by copy pasting the URL in my browser. 

I had suggested `python setup.py install` just to ensure you conda is synced with all changes you made in the local repo.

**Comment by saurabhclusterone** [comment]
@VIGS25 

I had run the latest code available from the repo once again after cloning the latest repo. the code failed with same error. I also ran the same code by cloning it on two other separate systems and they failed with same error.

Can you confirm once again please that all code changes have been checked into the repo.

Because I still that there is an indentation on line 24-25 in pdbbind_tf.py file in deepchem/examples/pdbbind directory.

**Comment by VIGS25** [comment]
@saurabhclusterone: 
A few points to be rechecked and confirmed:

1. The indentation issue is **NOT** in `pdbbind_tf.py` file but in `load_pdbbind_grid` function which can be found in `deepchem/molnet/load_function/pdbbind_datasets.py`. I hope we both agree to that.

2. I notice on your GitHub profile that your forked repository deepchem is 40 commits behind master. I hope you didn't use that repository for cloning.

3. Thirdly, the indentation error in `load_pdbbind_grid` which is in the `pdbbind_datasets.py` has been fixed and merged, 4 days ago.
(Have attached a screenshot of the same below)

![image](https://user-images.githubusercontent.com/32259379/47104834-9be5c000-d242-11e8-9843-7f2a486bfbfb.png)

**Comment by saurabhclusterone** [comment]
@VIGS25 

I understand what you have Witten. I just had one question:

If  have a conda environment and I install deepchem using following command as shown on Deepchem GitHub page:

conda install -c deepchem -c rdkit -c conda-forge -c omnia deepchem=2.1.0

Will it include the code changes that you had made or not?

**Comment by VIGS25** [comment]
@saurabhclusterone:
Ah, I'm not sure about this
I would recommend you install from source instead.

**Comment by saurabhclusterone** [comment]
@VIGS25 


Thank You for the reply! 
But, I wanted to know, if can we update this part in stable version please?

Otherwise, it is difficult to run it on AWS instances, where I will have to put extra scripts to install deepchem from source. To save this hassle, can we include at least this small change in the stable version?

**Comment by VIGS25** [comment]
I'm not well versed with the conda installation part, @rbharath would have a better answer on this!

**Comment by saurabhclusterone** [comment]
@rbharath

Could you please tell if we can merge the changes related to indentation error in pdbbind_datasets.py to the stable version of deepchem so that I can use deepchem by importing the library and not having to run it from source every time.

@VIGS25 had tagged you previously with regards to same. Look forward to hear from you. Thanks!

**Comment by rbharath** [comment]
Yes, absolutely can merge into stable version. All bugfixes in head will be included in the next release, DeepChem 2.2. This should be coming out in the next couple months.
